rust/lanzatool/src/generation.rs

```diff
@@ -1,47 +1,71 @@
+use serde::{Deserialize, Serialize};
+
 use std::fmt;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 use anyhow::{Context, Result};
+use bootspec::BootJson;
+use bootspec::SpecialisationName;
+use bootspec::generation::Generation;
+
+// TODO: actually, I'm not sure it's a good thing to have Default
+// we should maybe have TryDefault?
+// discuss this with upstream.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct SecureBootExtension {
+    #[serde(rename="org.lanzaboote.osRelease")]
+    pub os_release: PathBuf
+}
 
-use crate::bootspec::Bootspec;
+pub type ExtendedBootJson = BootJson<SecureBootExtension>;
 
 #[derive(Debug)]
-pub struct Generation {
+pub struct OSGeneration {
+    /// Top-level nixpkgs version
     version: u64,
-    specialisation_name: Option<String>,
-    pub bootspec: Bootspec,
+    /// Top-level specialisation name
+    specialisation_name: Option<SpecialisationName>,
+    /// Top-level bootspec document
+    pub bootspec: ExtendedBootJson,
+}
+
+fn into_boot_json(generation: Generation<SecureBootExtension>) -> Result<ExtendedBootJson> {
+    Ok(match generation {
+        Generation::V1(json) => json,
+        _ => panic!("Failed")
```

Comment:
a better error would be great hm

Reply:
Untested, but maybe something like this would be better?

```suggestion
        gen => return Err(format!("Unsupported generation: {:?}", gen))
```

Reply:
Yes, please don't panic here. There should be a human-readable and actionable error message, just like @cole-h suggests. :)

Reply:
Definitely agree, this was pure laziness here, thank you @cole-h for the suggestion